src/HttpClientRequestAdapter.cs

```diff
@@ -293,7 +295,70 @@ public ISerializationWriterFactory SerializationWriterFactory
                         {
                             result = rootNode.GetDateValue();
                         }
-                        else throw new InvalidOperationException("error handling the response, unexpected type");
+                        else if(
+                            Nullable.GetUnderlyingType(modelType) is { IsEnum: true } underlyingType &&

```

Comment:
We're going to need the AOT annotation on the generic type so people don't get errors when compiling with trimming.
https://github.com/microsoft/kiota-serialization-json-dotnet/blob/d0454daecb7ec2327f3e55d642fa510cf617e8b5/src/JsonParseNode.cs#L579

Reply:
The AOT attribute will also be required in the public method declaration I believe.

Reply:
Do you mean something like this?

`public async Task<ModelType?> SendPrimitiveAsync<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicFields)] ModelType>(RequestInformation requestInfo, Dictionary<string, ParsableFactory<IParsable>>? errorMapping = default, CancellationToken cancellationToken = default)`

Reply:
correct

Reply:
@andrueastman @baywet 
It's probably a breaking change which requires changes in Microsoft.Kiota.Abstractions

> error IL2095: 'DynamicallyAccessedMemberTypes' in 'DynamicallyAccessedMembersAttribute' on the generic parameter 'ModelType' of 'Microsoft.Kiota.Http.HttpClientLibrary.HttpClientRequestAdapter.SendPrimitiveAsync<ModelType>(RequestInformation, Dictionary<String, ParsableFactory<IParsable>>, CancellationToken)' don't match overridden generic parameter 'ModelType' of 'Microsoft.Kiota.Abstractions.IRequestAdapter.SendPrimitiveAsync<ModelType>(RequestInformation, Dictionary<String, ParsableFactory<IParsable>>, CancellationToken)'. All overridden members must have the same 'DynamicallyAccessedMembersAttribute' usage.

Is DynamicallyAccessedMembersAttribute necessary for SendPrimitiveAsync? Reflection is used on the underlying type, not directly on ModelType.

Reply:
I don't know the internals of AOT/trimming/the type system well enough to have a definitive answer answer to that question.
For changes impacting parse node and others, we did update the original interfaces, and that didn't result in a breaking change for anyone besides maybe people already doing AOT, but they were already broken.
I guess a first step here could be to look at the implementation/annotations on `Nullable.GetUnderlyingType(modelType)` and see what it requires (any dynamic attribute?) to know whether we need to propagate things here.

Reply:
At first sight, no dynamic attribute is used in the [implementation](https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Nullable.cs,ae2ae0d65dd71ecb)

Reply:
I think if we move the functionality to the abstractions side, things will become clearer/different. So, let's make that change first...